src/templates/plugin.main.js

```diff
@@ -126,6 +126,11 @@ export default async (context) => {
   }
 
   const doDetectBrowserLanguage = () => {
+    // Browser detection is ignored if it is a nuxt generate.
+    if (context.isStatic && context.ssrContext) {
```

Comment:
These are deprecated ways of checking generated/server mode. See https://nuxtjs.org/guide/plugins#client-side-only for suggested ones.

Reply:
I updated the check. Hope this is enough. 

Added a test that checks for file content. 
A direct call to '/fr/index.html' is necessary for the check.
Unfortunately I could not do this with the test browser.

Reply:
> Unfortunately I could not do this with the test browser.

What's the reason you couldn't do it? There is already test using `generate` there so I wonder why made it not feasible.

Reply:
OK, I see now. Because we are testing pre-rendered content.

Reply:
With the existing test there is no initial call of the non-default locale route.

If I open the /fr/index.html directly with a browser and would notice that it is empty.

With the test browser this case is unfortunately not to be adjusted, here the internal router seems to cover it.

It seems that if a fallback locale is defined in detectBrowserLanguage, the rendering of non-default local routes generates empty pages.